### ISSUE TITLE
USER ISSUE: Server Join Failure

### ISSUE THREAD

**Issue by egburr** [StrangeLoopGames/EcoIssues#4413]
**Version:** 0.7.1.0 beta

**Steps to Reproduce:**
Join server that I was on just an hour ago.

**Expected behavior:**
connect to server

**Actual behavior:**
The following error message popped up:
```
<size=150%>Connection Failed</size>

Caught exception: Caught exception invoking RPC ShouldSelectUser on Client!

Exception has been thrown by the target of an invocation.
Object reference not set to an instance of an object.


Stack:
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Object[] arguments, Signature sig, Boolean constructor)
   at System.Reflection.RuntimeMethodInfo.UnsafeInvokeInternal(Object obj, Object[] parameters, Object[] arguments)
   at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at Eco.Core.Controller.ControllerManager.PackageMethodResult(BSONObject bson, INetClient boundClient, IController controller, Tuple`2 method)
   at Eco.Core.Controller.ControllerManager.PackageController(IController controller, INetClient boundClient)
   at Eco.Core.Controller.ControllerManager.PackageValue(BSONObject bson, INetClient boundClient, Object value, Int32 nameID)
   at Eco.Core.Controller.ControllerManager.PackageController(IController controller, INetClient boundClient)
   at Eco.Core.Controller.ControllerManager.PackageValue(BSONObject bson, INetClient boundClient, Object value, Int32 nameID)
   at Eco.Core.Controller.ControllerManager.PackageController(IController controller, INetClient boundClient)
   at Eco.Plugins.Networking.Clients.Client.ShouldSelectUser()

```

AND
server recorded dump:
--BEGIN DUMP--
Dump Time
02/12/2018 15:11:43

Exception
System.Exception: Caught exception invoking RPC ShouldSelectUser on Client!

Exception has been thrown by the target of an invocation.
Object reference not set to an instance of an object.

Stack:
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Object[] arguments, Signature sig, Boolean constructor)
   at System.Reflection.RuntimeMethodInfo.UnsafeInvokeInternal(Object obj, Object[] parameters, Object[] arguments)
   at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at Eco.Core.Controller.ControllerManager.PackageMethodResult(BSONObject bson, INetClient boundClient, IController controller, Tuple`2 method)
   at Eco.Core.Controller.ControllerManager.PackageController(IController controller, INetClient boundClient)
   at Eco.Core.Controller.ControllerManager.PackageValue(BSONObject bson, INetClient boundClient, Object value, Int32 nameID)
   at Eco.Core.Controller.ControllerManager.PackageController(IController controller, INetClient boundClient)
   at Eco.Core.Controller.ControllerManager.PackageValue(BSONObject bson, INetClient boundClient, Object value, Int32 nameID)
   at Eco.Core.Controller.ControllerManager.PackageController(IController controller, INetClient boundClient)
   at Eco.Plugins.Networking.Clients.Client.ShouldSelectUser()

   at Eco.Shared.Networking.RPCManager.TryInvoke(Object controller, String methodname, BSONObject bsonArgs, Object& result)
   at Eco.Shared.Networking.RPCManager.InvokeOn(BSONObject bson, Object controller, String methodname)
   at Eco.Shared.Networking.RPCManager.HandleReceiveRPC(INetClient client, BSONObject bson)
   at Eco.Shared.Networking.NetPort.ProcessBuffer()
   at Eco.Shared.Networking.NetPort.Update()
   at Eco.Plugins.Networking.Clients.Client.<.ctor>b__55_0()

--END DUMP--

**Comment by egburr** [comment]
Some notes...
my username is "egburr" but while the steam link was down some days ago, when I logged on a user "egburr#2" was created. Ever y time I have logged on since then, I have had to select which one I want to use when I connect to this server. From the messages, it looks like this is related. This error pops up at the point where I am expecting the user selection pop-up to appear.

**Comment by egburr** [comment]
Also, I don't know if this is related, but AVG Anti-virus is occasionally popping up that it has protected me from URL:Mal.
Threat secured
We've safely aborted connection 62.102.148.187 because it was infected with URL:Mal.
URL: http://62.102.148.187:4764/info
Process: E:\SteamLibrary\SteamApps\common\Eco\Eco.exe

**Comment by MarjieAVolk** [comment]
Hey there - we are already working on a fix for this problem.  Should be fixed soon.  It's related to having used certain government or economic features on the server.

**Comment by egburr** [comment]
Thanks!

**Comment by egburr** [comment]
One more note....
I saw somewhere that game.eco is really a zip file, so I accessed it as a zip file, and found the user file for my "egburr#2" user and deleted it, and now I'm able to connect to the server again without this error. 

The only government/economic feature I think we've used so far is my friend created the jobs/contracts board (a big bulletin board looking thing) as a joke between us, as we've always worked together on these type of games anyway.

**Comment by MarjieAVolk** [comment]
Ah yeah.  If you had created a contract or something, then I could see how deleting your user would help.  Hopefully that works for you.  We've found that deleting your user can also break the save, for example if you had claimed property.

**Comment by egburr** [comment]
That user had not done anything.  No contracts or property. I only played him for maybe 10 minutes before logging off, and next time I logged on I guess the steam link was back, because ever since, I was given the choice of users to select from.

**Comment by MarjieAVolk** [comment]
Ah okay.  Glad the multiple users thing is working for you now :P  Yeah, deleting users can work sometimes, but we recently learned the hard way that it can also not work, so just warning you to be careful with it.

**Comment by egburr** [comment]
Oh, I backed up the entire "Storage" folder first, just in case.  :)

**Comment by metachronism** [comment]
Fixed in 7.2.  was a lingering server issue.
